### ISSUE TITLE
DocFX generates incorrect table markup for C# XMLDoc <list type="table"> tags

### ISSUE THREAD

**Issue by rjgotten** [dotnet/docfx#3614]
**DocFX Version Used**: 2.36.1.0

**Template used**:  `default`

**Steps to Reproduce**:

1. Create a C# class with XML API documentation that is supposed to render as a table, based on `<list type="table">` and associated markup. E.g. for the return value of an `IComparer<T>` class:
```csharp
/// <returns>
/// A signed integer that indicates the relative value of <paramref name="x"/>
/// and <paramref name="y"/>, as shown in the following table.
/// <list type="table">
///   <listheader>
///     <term>Value</term>
///     <term>Meaning</term>
///   </listheader>
///   <item>
///     <description>Less than zero</description>
///     <description><paramref name="x"/> is less than <paramref name="y"/>.</description>
///   </item>
///   <item>
///     <description>Zero</description>
///     <description>when <paramref name="x"/> equals <paramref name="y"/>.</description>
///   </item>
///   <item>
///     <description>Greater than zero</description>
///     <description><paramref name="x"/> is greater than <paramref name="y"/>.</description>
///   </item>
/// </list>
/// </returns>
public int Compare(TSource x, TSource y)
```
2. Generate DocFX based documentation for this class.

**Expected Behavior**:
A nicely formatted table is generated, where the `<term>` cells form the column header cells and the `<description>` cells form the normal data rows:
```html
<table>
  <thead>
    <tr><th>Value</th><th>Meaning</th></tr>
  </thead>
  <tbody>
    <tr><td>Less than zero</td><td><code>x</code> is less than <code>y</code>.</td>
    <tr><td>Zero</td><td><code>x</code> equals <code>y</code>.</td>
    <tr><td>Greater than zero</td><td><code>x</code> is more than <code>y</code>.</td>
  </tbody>
</table>
```

**Actual Behavior**:
An obviously incorrect mess is generated. All `<term>` nodes are grouped together into the first `<th>` and all `<description>` nodes are grouped together into the last `<td>`:

```html
<table>
  <thead>
    <tr><th>ValueMeaning</th><th></th></tr>
  </thead>
  <tbody>
    <tr><td></td><td>Less than zero<code>x</code> is less than <code>y</code>.</td>
    <tr><td></td><td>Zero<code>x</code> equals <code>y</code>.</td>
    <tr><td></td><td>Greater than zero<code>x</code> is more than <code>y</code>.</td>
  </tbody>
</table>
```

Hazarding a guess here; whomever was responsible for implementing this code is making some very false assumptions on how `<list>` tags work and simply short-circuits the entire thing to only work as a definition list, where `<term>` always goes into the first column and `<description>` always goes into the second. Which seems to be the incorrect way to handle these tags.

The [published documentation for `<list>`](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/list) states that for tables, you need only supply an entry for `<term>` in the list header and an entry for `<description>` in items.

So, what's the expectation here?

**Comment by herohua** [comment]
@rjgotten , thanks for bringing this up. It looks like something not considered when the feature is implemented. We will put this request in backlog.

**Comment by rjgotten** [comment]
@herohua 
Should your bot be marking backlogged requests as stale?

**Comment by superyyrrzz** [comment]
@rjgotten Sorry for confusing you due to some bot rule issue. This issue is safe now as it has the `feature-request` label.

**Comment by Tape-Worm** [comment]
Is there any eta on this?  All of my tables look horrible.

**Comment by Tape-Worm** [comment]
Just FYI, if anyone's even still watching this, the fix is really simple.

In the TripleSlashCommentTransform.xsl there's a block that generates the table:
```
<thead>
  <tr>
    <th>
      <xsl:apply-templates select="listheader/term" />
    </th>
    <th>
      <xsl:apply-templates select="listheader/description" />
    </th>
  </tr>
</thead>
</xsl:if
<tbody>
  <xsl:for-each select="item">
      <tr>
        <td>
          <xsl:apply-templates select="term"/>
        </td>
        <td>
        <xsl:apply-templates select="description"/>
        </td>
    </tr>
  </xsl:for-each>
</tbody>
```

To fix this, we need to remove the `<xsl:apply-templates select="..."/>`  from the `<th>` and `<td>` tags and replace them with `<xsl:apply-templates />`

Then, before each `<th>` and `<td>`we need to add `<xsl:for-each select="listheader/term" />` (or `<xsl:for-each select="listheader/description" />` depending on the case) and `</xsl:for-each>` after each `</th>` and `</td>`

Here's the fixed code:
```
<thead>
  <tr>
	<xsl:for-each select="listheader/term">
	<th>
	  <xsl:apply-templates />
	</th>
	</xsl:for-each>
	<xsl:for-each select="listheader/description">
	<th>
	  <xsl:apply-templates />
	</th>
	</xsl:for-each>
  </tr>
</thead>
</xsl:if>
<tbody>
<xsl:for-each select="item">
  <tr>
	<xsl:for-each select="term">
	<td>
	  <xsl:apply-templates />
	</td>
	</xsl:for-each>
	<xsl:for-each select="description">
	<td>
	  <xsl:apply-templates />
	</td>
	</xsl:for-each>
  </tr>
</xsl:for-each>
</tbody>
```

Hopefully this expedites things.

I don't want to really jump through the hoops required to put up a PR for this, so if someone else wants to, be my guest.

**Comment by KalleOlaviNiemitalo** [comment]
This seems to be a duplicate of <https://github.com/dotnet/docfx/issues/6864>, which was fixed in DocFX 2.56.6 with PR <https://github.com/dotnet/docfx/pull/6865>. I did not test, though.

**Comment by rjgotten** [comment]
Technically #6864 is the duplicate, since it came later. But let's not split hairs over that.
I'm happy to see this fixed. It _does_ look like it's actually the same issue.
